build-aux/m4/l_atomic.m4

```diff
@@ -35,16 +35,22 @@ AC_DEFUN([CHECK_ATOMIC], [
 
   AC_LANG_PUSH(C++)
 
+  LIBS_TEMP="$LIBS"
+  unset LIBS
+  LIBS="$LIBS_TEMP $PTHREAD_CFLAGS $PTHREAD_LIBS"
```

Comment:
The `PTHREAD_CFLAGS` flags are only required. Suggesting to use them in a more idiomatic way:
```diff
--- a/build-aux/m4/l_atomic.m4
+++ b/build-aux/m4/l_atomic.m4
@@ -18,7 +18,7 @@ m4_define([_CHECK_ATOMIC_testbody], [[
 
   int main() {
     std::atomic<bool> lock{true};
-    std::atomic_exchange(&lock, false);
+    lock.exchange(false);
 
     std::atomic<std::chrono::seconds> t{0s};
     t.store(2s);
@@ -34,6 +34,8 @@ m4_define([_CHECK_ATOMIC_testbody], [[
 AC_DEFUN([CHECK_ATOMIC], [
 
   AC_LANG_PUSH(C++)
+  TEMP_CXXFLAGS="$CXXFLAGS"
+  CXXFLAGS="$CXXFLAGS $PTHREAD_CFLAGS"
 
   AC_MSG_CHECKING([whether std::atomic can be used without link library])
 
@@ -51,5 +53,6 @@ AC_DEFUN([CHECK_ATOMIC], [
         ])
     ])
 
+  CXXFLAGS="$TEMP_CXXFLAGS"
   AC_LANG_POP
 ])
```

Reply:
I've Guix build a [branch with this simplification](https://github.com/fanquake/bitcoin/commit/67fc2ead2dd0fefe88a86fc3fd4049db244d1d6b), and it's working as expected. @achow101 do you want to incorporate this change?

Guix Build (x86_64):
```bash
88a10edec62a0ffaee9c209076d50a83119d01d3242b4bde147d5936bd9f2dd3  guix-build-67fc2ead2dd0/output/dist-archive/bitcoin-67fc2ead2dd0.tar.gz
36d9d889df9ec53d0926b3c7e8d5b8aec3ade2c9812ce91b747f6b0cd02917b4  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/SHA256SUMS.part
3078cc2e8e4c420b13a363815ef560e6d3c7a59d4f3b19c083a4ae5141094d68  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/bitcoin-67fc2ead2dd0-riscv64-linux-gnu-debug.tar.gz
ac70527f4492c18fc0eb0e3f0eaac40bc5264f621cf82c2267c40580e9fc30f0  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/bitcoin-67fc2ead2dd0-riscv64-linux-gnu.tar.gz
```

Guix Build (arm64):
```bash
88a10edec62a0ffaee9c209076d50a83119d01d3242b4bde147d5936bd9f2dd3  guix-build-67fc2ead2dd0/output/dist-archive/bitcoin-67fc2ead2dd0.tar.gz
36d9d889df9ec53d0926b3c7e8d5b8aec3ade2c9812ce91b747f6b0cd02917b4  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/SHA256SUMS.part
3078cc2e8e4c420b13a363815ef560e6d3c7a59d4f3b19c083a4ae5141094d68  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/bitcoin-67fc2ead2dd0-riscv64-linux-gnu-debug.tar.gz
ac70527f4492c18fc0eb0e3f0eaac40bc5264f621cf82c2267c40580e9fc30f0  guix-build-67fc2ead2dd0/output/riscv64-linux-gnu/bitcoin-67fc2ead2dd0-riscv64-linux-gnu.tar.gz
```

Reply:
I had some additional uneeded code left in that branch. Have force pushed to drop it, and done new builds.
Guix Build (x86_64):
```bash
a79f45073561315f06a2123a0f77d8a2d151beee3a21b4fbacc79c88b7068ea1  guix-build-fbf863560bc3/output/dist-archive/bitcoin-fbf863560bc3.tar.gz
04666a68e548e2965fe2bfcb80ef3851671f90b2e06b9cebaca2b8eba585b9dd  guix-build-fbf863560bc3/output/riscv64-linux-gnu/SHA256SUMS.part
172b5b301dcb3edd8bb562fd3d3522d8a18ccf69aa50e87723202dd5b1c8bc2a  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu-debug.tar.gz
40df37725534781b81ac556be79ef54fc350de1f9fb6a88de182ea5a98bc6977  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu.tar.gz
```

Guix Build (arm64):
```bash
a79f45073561315f06a2123a0f77d8a2d151beee3a21b4fbacc79c88b7068ea1  guix-build-fbf863560bc3/output/dist-archive/bitcoin-fbf863560bc3.tar.gz
04666a68e548e2965fe2bfcb80ef3851671f90b2e06b9cebaca2b8eba585b9dd  guix-build-fbf863560bc3/output/riscv64-linux-gnu/SHA256SUMS.part
172b5b301dcb3edd8bb562fd3d3522d8a18ccf69aa50e87723202dd5b1c8bc2a  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu-debug.tar.gz
40df37725534781b81ac556be79ef54fc350de1f9fb6a88de182ea5a98bc6977  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu.tar.gz
```

Reply:
#### Guix builds on `x86_64`:
```
a79f45073561315f06a2123a0f77d8a2d151beee3a21b4fbacc79c88b7068ea1  guix-build-fbf863560bc3/output/dist-archive/bitcoin-fbf863560bc3.tar.gz
04666a68e548e2965fe2bfcb80ef3851671f90b2e06b9cebaca2b8eba585b9dd  guix-build-fbf863560bc3/output/riscv64-linux-gnu/SHA256SUMS.part
172b5b301dcb3edd8bb562fd3d3522d8a18ccf69aa50e87723202dd5b1c8bc2a  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu-debug.tar.gz
40df37725534781b81ac556be79ef54fc350de1f9fb6a88de182ea5a98bc6977  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu.tar.gz
```

Reply:
Guix builds on `arm64` for @fanquake's [branch](https://github.com/fanquake/bitcoin/commit/fbf863560bc3b129d6e258becf410ef4ffbfd123):
```
a79f45073561315f06a2123a0f77d8a2d151beee3a21b4fbacc79c88b7068ea1  guix-build-fbf863560bc3/output/dist-archive/bitcoin-fbf863560bc3.tar.gz
04666a68e548e2965fe2bfcb80ef3851671f90b2e06b9cebaca2b8eba585b9dd  guix-build-fbf863560bc3/output/riscv64-linux-gnu/SHA256SUMS.part
172b5b301dcb3edd8bb562fd3d3522d8a18ccf69aa50e87723202dd5b1c8bc2a  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu-debug.tar.gz
40df37725534781b81ac556be79ef54fc350de1f9fb6a88de182ea5a98bc6977  guix-build-fbf863560bc3/output/riscv64-linux-gnu/bitcoin-fbf863560bc3-riscv64-linux-gnu.tar.gz
```